src/casp/services/api/services/cell_operations_service.py

```diff
@@ -60,36 +60,53 @@ def get_cache_info(user: models.User) -> schemas.CacheInfo:
             module_cache_info=cache_info["module_cache_info"],
         )
 
-    def __get_cell_count_from_anndata(self, file: t.BinaryIO) -> int:
+    def __validate_anndata_type(self, adata: anndata.AnnData):
```

Comment:
I think it's a great call to have this function. I think it'd be also important to include the check for`total_mrna_umis` ( if it is in `adata["obs"]`). It's a computationally cheap check, but it's crucial to have it as without it, model_service would raise an error.

Reply:
Ooo that's a good idea.  I'll add that